### PR TITLE
System for beds, Fog lore book added, adventurer paladin dupefix

### PR DESCRIPTION
**Pull request by NPC1314**
## About The Pull Request

While fog and rain remains broken, I had an epiphany that actually kinda makes sense for why our fog is so derpy.
Added a fog lore book to the pool.

Also cleans up & repaths beds so instead of  whacky duplications and sub-categories you get four actually functional classes of beds, inn beds with qual 3, wool bed with qual 2, normal beds with qual 1 and bad beds with qual 0.5

So you can map shit beds in accessible areas, inn ones only for King and Priest and Inn, wool for a few nobles and rest mediocre or something.

Also cleans up some missed pathing and a spell dupe for paladins.

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

### CODE REVIEW

**Comment by BadAtThisGame302** [comment]
The's also rain indoors and it doesn't fill up buckets/clean blood but that isn't removed. Don't remove fog and not replace it with another weather event. People hate the Fog because they should hate it, it's a hindrance.

File: code/game/objects/structures/beds_chairs/roguechair.dm [hunk 1]
```diff
@@ -290,19 +290,91 @@
...
+/obj/structure/bed/rogue/inn
+	icon_state = "inn_bed"
 	sleepy = 3
+
+/obj/structure/bed/rogue/inn/double
+	icon_state = "double"
+	pixel_y = 0
+	sleepy = 3
+	debris = list(/obj/item/grown/log/tree/small = 2)
+/*            ///////WIP  This will essentially allow for multiple mobs to buckle, just needs to change mousedrop function
```
Comment (BadAtThisGame302) [inline code/game/objects/structures/beds_chairs/roguechair.dm:318]: Tried getting this to work?

Comment (NPC1314) [reply code/game/objects/structures/beds_chairs/roguechair.dm:318]: nope, old comment, could take a look at it later to see whats wrong with it

**Comment by NPC1314** [comment]
What if...its not fog.....not natural at least....its vapors leaking up from Necras underworld...a mystical fog literally streaming from the ground, ghost fog....that way it makes sense

**Comment by ThePainkiller** [comment]
If you do please comment it out, I might just refactor this weather to lavaland variants and make it a per-tile based like lavaland and say fuck it, but we might need proper loopable resprites for this effect...

**Comment by bowlofcereal** [comment]
> The's also rain indoors and it doesn't fill up buckets/clean blood but that isn't removed. Don't remove fog and not replace it with another weather event. People hate the Fog because they should hate it, it's a hindrance.

Funny enough last round I played the rain *did* actually fill up water buckets. It was the first time I had ever seen it work and it was like seeing a unicorn.

File: code/game/objects/structures/beds_chairs/roguechair.dm (hunk 1 above)
Comment (NPC1314) [reply code/game/objects/structures/beds_chairs/roguechair.dm]: the comment is correct, it works but need to some coding changing how the mousedrag works, too much work.

**Review by Optimism333 — COMMENTED** [review]
Once the conflict is resolved, can go in for a test.

**Review by Optimism333 — APPROVED** [review]

**Pull request merged**